lib/galaxy/webapps/galaxy/api/folder_contents.py

```diff
@@ -2,93 +2,99 @@
 API operations on the contents of a library folder.
 """
 import logging
+from typing import Optional
 
-from galaxy import util
-from galaxy.web import (
-    expose_api,
-    expose_api_anonymous,
+from fastapi import (
+    Body,
+    Path,
+    Query,
+)
+
+from galaxy.managers.context import ProvidesUserContext
+from galaxy.schema.fields import DecodedDatabaseIdField
+from galaxy.schema.schema import (
+    CreateLibraryFilePayload,
+    LibraryFolderContentsIndexQueryPayload,
+    LibraryFolderContentsIndexResult,
 )
 from galaxy.webapps.galaxy.services.library_folder_contents import LibraryFolderContentsService
 from . import (
-    BaseGalaxyAPIController,
     depends,
+    DependsOnTrans,
+    Router,
 )
 
 log = logging.getLogger(__name__)
 
+router = Router(tags=["data libraries folders"])
 
-class FolderContentsController(BaseGalaxyAPIController):
-    """
-    Class controls retrieval, creation and updating of folder contents.
-    """
-
-    service: LibraryFolderContentsService = depends(LibraryFolderContentsService)
-
-    @expose_api_anonymous
-    def index(self, trans, folder_id, limit=None, offset=None, search_text=None, **kwd):
-        """
-        GET /api/folders/{encoded_folder_id}/contents?limit={limit}&offset={offset}
-
-        Displays a collection (list) of a folder's contents
-        (files and folders). Encoded folder ID is prepended
-        with 'F' if it is a folder as opposed to a data set
-        which does not have it. Full path is provided in
-        response as a separate object providing data for
-        breadcrumb path building.
-
-        ..example:
-            limit and offset can be combined. Skip the first two and return five:
-                '?limit=3&offset=5'
-
-        :param  folder_id: encoded ID of the folder which
-            contents should be library_dataset_dict
-        :type   folder_id: encoded string
-
-        :param  offset: offset for returned library folder datasets
-        :type   folder_id: encoded string
-
-        :param  limit: limit   for returned library folder datasets
-            contents should be library_dataset_dict
-        :type   folder_id: encoded string
-
-        :param kwd: keyword dictionary with other params
-        :type  kwd: dict
+FolderIdPathParam: DecodedDatabaseIdField = Path(
+    ..., title="Folder ID", description="The encoded identifier of the library folder."
```

Comment:
I don't know/understand much of the pydantic stuff, but I'd expect to find EncodedDatabaseIdField at the API layer like here, and DecodedDatabaseIdField in services/managers/controllers. Here it looks confusing when you compare the type and description.

Reply:
I got a bit confused too at the beginning, that's why I kept using `EncodedDatabaseIdField`, but using `DecodedDatabaseIdField` is waaaay cleaner as Marius suggested, we can avoid any further explicit decoding down the road.
I guess we can just rename it to `DatabaseIdField` or something similar to reduce the confusion after we get rid of the existing `EncodedDatabaseIdField`s.